### PR TITLE
As tm items purchased

### LINKED ISSUE #9: 8. As a user, I want to mark an item on my shopping list as purchased so the app can learn how often I buy different items.
AC:

* User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
* Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

### PR DESCRIPTION
**Pull request by tarynblakemiller**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

1. We used React Hook's UseState and UseEffect methods in tandem with onChange events in order to set the state of the items' associated checkboxes, and determine whether or not they should remain checked as well as appropriately update the Firestore. 

## Related Issue

Closes #9 

## Acceptance Criteria

1. User is able to tap a checkbox or similar UI element to mark an item in the list as purchased

2. Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

Alt Text

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin as-tm-items-purchased and check into that branch with git checkout as-tm-items-purchased

- Run npm start to launch the server

- Check an item that you want to be marked as "purchased", and check the item's purchaseDate field in the Firestore to see the changes. 

- When you check that item again, the purchaseDate should now be reset to null.

### CODE REVIEW

File: src/components/GroceryItem/GroceryItem.js [hunk 1]
```diff
@@ -1,9 +1,54 @@
...
+import { firestore } from '../../lib/firebase.js';
 
 const GroceryItem = ({ item }) => {
+  const groceryItem = firestore.collection('groceryItems').doc(item.id);
+  const [isChecked, setChecked] = useState(false);
+
+  const oneday = 60 * 60 * 24 * 1000;
+
+  useEffect(() => {
+    if (item.purchaseDate > Date.now() - oneday) {
```
Comment (vanmars) [inline src/components/GroceryItem/GroceryItem.js:11]: Clever way to check an item as purchased for only 24 hours after it was bought!

**Review by vanmars — APPROVED** [review]
Looks great! Nice work!

**Review by CodenameTam — APPROVED** [review]

File: src/components/GroceryItem/GroceryItem.js [hunk 2]
```diff
@@ -1,9 +1,54 @@
...
+      setChecked(true);
+    }
+  }, []);
+
+  const setNewDate = () => {
+    return groceryItem
+      .update({
+        purchaseDate: Date.now(),
+      })
+      .then(() => {
```
Comment (lyleschemmerling) [inline src/components/GroceryItem/GroceryItem.js:21]: We are using async/await in other parts of the code. Not necessary but would consider using it here `try { await groceryItem.update... } catch (error) {/...`

**Comment by lyleschemmerling** [comment]
Looks great!

**Review by lyleschemmerling — APPROVED** [review]

**Pull request merged**